tests/e2e/config_secret_utils.go

```diff
@@ -800,3 +804,43 @@ func createTestUserAndAssignLimitedRolesAndPrivileges(masterIp string, sshClient
 		}
 	}
 }
+
+// verifyClusterIdConfigMapGeneration verifies if cluster id configmap gets generated by
+// csi driver in csi namespace
+func verifyClusterIdConfigMapGeneration(client clientset.Interface, ctx context.Context,
+	csiNamespace string, cmToExist bool) {
+	_, err := client.CoreV1().ConfigMaps(csiNamespace).Get(ctx, vsphereClusterIdConfigMapName,
+		metav1.GetOptions{})
+	if cmToExist && apierrors.IsNotFound(err) {
+		framework.Logf("Configmap: %s not found in namespace: %s", vsphereClusterIdConfigMapName, csiNamespace)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	} else if !cmToExist && !apierrors.IsNotFound(err) {
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	}
+}
+
+// fetchClusterIdFromConfigmap fetches cluster id value from
+// auto generated cluster id configmap by csi driver
+func fetchClusterIdFromConfigmap(client clientset.Interface, ctx context.Context,
+	csiNamespace string) string {
+	clusterIdCm, err := client.CoreV1().ConfigMaps(csiNamespace).Get(ctx, vsphereClusterIdConfigMapName,
+		metav1.GetOptions{})
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	data := clusterIdCm.Data
+	framework.Logf("cluster id configmap: %v", clusterIdCm)
+	return data["clusterID"]
+}
+
+// recreateVsphereConfigSecret recreates config secret with new config parameters
+// and restarts CSI driver
+func recreateVsphereConfigSecret(client clientset.Interface, ctx context.Context,
```

Comment:
why we are writing new util for config secret creation? 
Is this method createCsiVsphereSecret not meeting your testcode requirements?